src/App.tsx

```diff
@@ -21,11 +44,13 @@ export const App: React.FC = () => {
             data-cy="perPageSelector"
             id="perPageSelector"
             className="form-control"
+            value={perPage}
+            onChange={(event) => setPerPage(Number(event.target.value))}
           >
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            <option value={3}>3</option>
+            <option value={5}>5</option>
+            <option value={10}>10</option>
+            <option value={20}>20</option>
```

Comment:
You don't need to change values to number here, because you do it in 48 line in your code, when you're getting value from event.target.